### DISCUSSION TITLE
Splitting a long lists of targets

### DISCUSSION THREAD

**Discussion by llrs** [ropensci/targets#1018] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I think I saw some users splitting a project in several lists and then combining them together in a single pipeline. I tried looking at stackoverflow and other sites but searching by keyword didn't result in great matches. I think I saw a gist some years ago with it, but I don't remember the user or the topic to search specifically for that content. 

I've read the [projects' chapter](https://books.ropensci.org/targets/projects.html#interdependent-projects) of the book, and I don't know if this feature(?) has been superseeded by the support for multiple projects and interdependent projects (which I am also considering, depending on the growing & changing needs of some projects). 

Any examples might be appreciated!

**Comment by wlandau** [comment]
It is more of a programming technique than an explicit feature. `tar_target()` returns an object, and you can create target objects wherever you need to. You can have a pipeline like this:

```r
# _targets.R
source("R/functions.R")
source("R/first_targets.R")
source("R/last_targets.R")
list(list_of_first_targets, list_of_last_targets)
```

where these target lists are defined in helper scripts:

```r
# R/first_targets.R
list_of_first_targets <- list(
  tar_target(raw_data, get_raw_data()),
  tar_target(data, preprocess_data(raw_data)
)
```

```r
# R/last_targets.R
list_of_last_targets <- list(
  tar_target(model, fit_model(data)),
  tar_target(summary, summarize_model(model)
)
```

You can split up your targets any way you like into nested lists, not necessarily in topological order.

`tar_target()` returns a classed environment with special instructions and metadata inside.

```r
object <- tar_target(x, 1)
is.environment(object)
#> [1] TRUE
class(object)
#> [1] "tar_stem"    "tar_builder" "tar_target" 
#> [4] "environment"
```

**Reply by llrs** [reply]
Many thanks! I think I tried with this and I got some errors (but it could have been something else, or just using `c()` instead of `list()` to combine them). Wouldn't it be better to use `tar_source()` to load all at the same time?

**Reply by wlandau** [reply]
`tar_source()` is better, I used `source()` just to show the connections among the scripts.

**Reply by llrs** [reply]
Many for your fast help!!
